Duplicate tags lead to unexpected behaviour

**Environment information**
* MWDB version (from `/about`): 2.6.1
* Installation method:
  - [ ] mwdb.cert.pl service
  - [x] From PyPi (`pip install mwdb-core`)
  - [ ] From docker-compose
  - [ ] Other (please explain)
* Plugins installed:

**Behaviour the bug (what happened?)**

Using mwdblib I used `upload_file()` to upload a file which worked based on the logs:

```
[INFO] Thread-195 - object.create_object:132 - File added - dhash:647fa66c2cef019a16ae46d3648949b6777fb2f03ceeff8494ae76e65b34f187 - is_new:True
[INFO] Thread-195 - app.log_request:140 - request - path:/api/file - arguments:{} - method:POST - status:200 - response_time:0:00:00.087994 - response_size:1109
127.0.0.1 - - [10/Apr/2022 15:33:54] "POST /api/file HTTP/1.0" 200 -
```

I then proceeded to upload a blob:

```
[INFO] Thread-199 - object.create_object:132 - TextBlob added - dhash:9b2e296c12e3490f6e0d5b185a10f6e87100ebb92037568307f22c361588efa3 - is_new:True
[INFO] Thread-199 - app.log_request:140 - request - path:/api/blob - arguments:{} - method:POST - status:200 - response_time:0:00:00.023690 - response_size:49679
127.0.0.1 - - [10/Apr/2022 15:33:55] "POST /api/blob HTTP/1.0" 200 -
```

Based on the `is_new:True` and HTTP 200 I reckoned this worked fine. However when I then attempted to add the blob as a child to the parent:

```
[ERROR] Thread-200 - service.error_router:71 - 404 Not Found: Child object not found
[INFO] Thread-200 - app.log_request:140 - request - path:/api/object/647fa66c2cef019a16ae46d3648949b6777fb2f03ceeff8494ae76e65b34f187/child/9b2e296c12e3490f6e0d5b185a10f6e87100ebb92037568307f22c361588efa3 - arguments:{} - method:PUT - status:404 - response_time:0:00:00.015540 - response_size:44
127.0.0.1 - - [10/Apr/2022 15:33:55] "PUT /api/object/647fa66c2cef019a16ae46d3648949b6777fb2f03ceeff8494ae76e65b34f187/child/9b2e296c12e3490f6e0d5b185a10f6e87100ebb92037568307f22c361588efa3 HTTP/1.0" 404 -
```

The child (blob) isn't found. This left me puzzled until I realized the `upload_blob()` call had a list of tags where two tags where duplicates. Pruning the duplicate and submitting again worked as epxected.

**Expected behaviour**

<!-- What did you expect to happen? -->
Either mwdblib would treat the `tags` as a `set()` or have another method of removing duplicates.
Or perhaps more appropriately, mwdb-core would return an error when an object is added with duplicate tags.

**Reproduction Steps**

Something along these lines mimics my original scenario:

```python
file = mwdblib.upload_file(name='sample file', content=b'dummy file')
blob = mwdlib.upload_blob(name='sample blob', content=b'dummy blob', tags=['tag1', 'tag2', 'tag1'])
file.add_child(blob.id)
```

--- a/mwdb/model/object.py
+++ b/mwdb/model/object.py
@@ -697,13 +697,15 @@ class Object(db.Model):
         db_tag.tag = tag_name
         db_tag, _ = Tag.get_or_create(db_tag)
 
+        if db_tag in self.tags:
+            return False
+
         is_new = False
         db.session.begin_nested()
         try:
-            if db_tag not in self.tags:
-                self.tags.append(db_tag)
-                db.session.commit()
-                is_new = True
+            self.tags.append(db_tag)
+            db.session.commit()
+            is_new = True
         except IntegrityError:
             db.session.rollback()
             db.session.refresh(self)
@@ -726,13 +728,15 @@ class Object(db.Model):
         else:
             db_tag = db_tag.one()
 
+        if db_tag not in self.tags:
+            return False
+
         is_removed = False
         db.session.begin_nested()
         try:
-            if db_tag in self.tags:
-                self.tags.remove(db_tag)
-                db.session.commit()
-                is_removed = True
+            self.tags.remove(db_tag)
+            db.session.commit()
+            is_removed = True
         except IntegrityError:
             db.session.rollback()
             db.session.refresh(self)
@@ -969,13 +973,15 @@ class Object(db.Model):
         if db_analysis is None:
             return False
 
+        if db_analysis not in self.analyses:
+            return False
+
         is_removed = False
         db.session.begin_nested()
         try:
-            if db_analysis in self.analyses:
-                self.analyses.remove(db_analysis)
-                db.session.commit()
-                is_removed = True
+            self.analyses.remove(db_analysis)
+            db.session.commit()
+            is_removed = True
         except IntegrityError:
             db.session.rollback()
             db.session.refresh(self)

